### ISSUE TITLE
Downloads from Oracle failing again

### ISSUE THREAD

**Issue by lae** [srsp/ansible-oracle-java#2]
Just ran a periodic build for one of my roles and it seems it's failing when attempting to download a JRE:

https://travis-ci.org/lae/ansible-role-elasticsearch/jobs/289276963#L4473

I didn't get much time to look at it but it looks like Oracle may have changed something with regards to fetching the JRE? 8u144 seems to still be the most recent JRE 8 version (though I guess it might make sense to change the default to JRE 9 maybe).

**Comment by lae** [comment]
https://github.com/William-Yeh/ansible-oracle-java/issues/62#issuecomment-336179446 might be relevant

**Comment by yuppie-flu** [comment]
My deployment was also blocked by this issue. I made small investigation. Here are the results:
It happened because Oracle release new update of Java 8 including some security fixes. The previous version was deprecated. Check details here:
http://www.oracle.com/technetwork/java/javase/8u152-relnotes-3850503.html
http://www.oracle.com/technetwork/java/javase/8u151-relnotes-3850493.html

I solved the issue by specifying the following variables:
```
java_version: 8
java_subversion: 152
java_build: 16
jdk_tarball_hash: aa0333dd3019491ca4f6ddbe78cdb6d0
```
If you want 151 subversion, find appropriate values for build and tarball hash here:
http://www.oracle.com/technetwork/java/javase/downloads/jdk8-downloads-2133151.html

**Comment by lae** [comment]
Ah okay, so it was just an update. My coworker must have been looking at the wrong thing (it looked like the latest was still 8u144). Anyway, looks like you opened #3 so 👍

**Comment by srsp** [comment]
That's interesting. 8u141 still seems to work. Normally Oracle does not pull the old versions, but 8u144 is gone. So 8u152 seems like a reasonable default now. 

I would not make Java 9 the default, because that might break a few things. Most projects I tried to build with Java 9 did not work out of the box because of the changes Jigsaw introduced. That does not mean Java 9 will never be the default, but I think it makes sense to wait a bit.

**Comment by srsp** [comment]
Fixed with #3 thanks to @yuppie-flu
